src/main/java/dk/cngroup/lentils/controller/ProgressController.java

```diff
@@ -40,36 +38,33 @@ public String listProgress(Model model) {
     @GetMapping(value = "/stage")
     public String stageProgress(@RequestParam("cypherId") Long cypherId, Model model) {
         Cypher cypher = cypherService.getCypher(cypherId);
-        fillModelAttributes(model, teamService.getAll(), cypher);
-        return PROGRESS_STAGE;
-    }
-
-    @GetMapping(value = "/skip/{cypherId}")
-    public String stageSkip (@PathVariable("cypherId") Long cypherId, @RequestParam("teamId") Long teamId, Model model){
-        model.addAttribute("status", progressService.setSkipProgressStatus(cypherService.getCypher(cypherId),
-                teamService.getTeam(teamId)));
-        fillModelAttributes(model, teamService.getAll(), cypherService.getCypher(cypherId));
-        return PROGRESS_STAGE;
-    }
-
-    @GetMapping(value = "/pass/{cypherId}")
-    public String stagePass (@PathVariable("cypherId") Long cypherId, @RequestParam("teamId") Long teamId, Model model){
-        model.addAttribute("status", progressService.setPassProgressStatus(cypherService.getCypher(cypherId),
-                teamService.getTeam(teamId)));
-        fillModelAttributes(model, teamService.getAll(), cypherService.getCypher(cypherId));
-        return PROGRESS_STAGE;
-    }
-
-    @GetMapping(value = "/pending/{cypherId}")
-    public String stagePending (@PathVariable("cypherId") Long cypherId, @RequestParam("teamId") Long teamId, Model model){
-        model.addAttribute("status", progressService.setPendingProgressStatus(cypherService.getCypher(cypherId),
-                teamService.getTeam(teamId)));
-        fillModelAttributes(model, teamService.getAll(), cypherService.getCypher(cypherId));
+        model.addAttribute("cypher", cypher);
+        model.addAttribute("teams", teamService.getAll());
+        model.addAttribute("teamsStatus", progressService.getTeamsStatuses(cypher));
         return PROGRESS_STAGE;
     }
 
-    private void fillModelAttributes(Model model, List<Team> teams, Cypher cypher) {
-        model.addAttribute("teams", teams);
+    @GetMapping(value = "/changeStatus/{cypherId}")
+    public String changeCypherStatus (@PathVariable("cypherId") Long cypherId,
+                                      @RequestParam("teamId") Long teamId,
+                                      @RequestParam("newStatus") String newStatus,
+                                      Model model) {
+        Cypher cypher = cypherService.getCypher(cypherId);
+        Team team = teamService.getTeam(teamId);
+        switch (newStatus) {
```

Comment:
Hmmm tady bych to zrefaktoroval. Idealni by bylo misto switche tady jen zavolat neco jako create metodu tovarny vytvareni status enumu ze stringu a pak zavolat jen progressStatus.setCypherStatus(cypher, team, status). Tim padem tady nebude ten velky switch a zaroven bude jen jedna metoda v progress service. Ta metoda pak bude muset byt natolik obecna, aby delala i tu validaci a ty podminky te validace by mely byt v tom Status enumu. V praci si to ukazem vic a vysvetlim ti to vic :)